packages/types/src/codec/U8aFixed.ts

```diff
@@ -2,43 +2,44 @@
 // This software may be modified and distributed under the terms
 // of the ISC license. See the LICENSE file for details.
 
-import { AnyU8a } from '../types';
+import toU8a from '@polkadot/util/u8a/toU8a';
 
+import { AnyU8a } from '../types';
 import U8a from './U8a';
 
 type BitLength = 8 | 16 | 32 | 64 | 128 | 256 | 512;
 
 // A U8a that manages a a sequence of bytes up to the specified bitLength. Not meant
 // to be used directly, rather is should be subclassed with the specific lengths.
 export default class U8aFixed extends U8a {
-  protected _bitLength: number;
+  protected _bitLength: BitLength;
 
   constructor (value: AnyU8a = new Uint8Array(), bitLength: BitLength = 256) {
-    super(value);
+    super(
+      value instanceof U8aFixed
+        ? value
+        : U8aFixed.trimLength(toU8a(value), bitLength / 8)
```

Comment:
Just check current master as well, I had some hoops to jump through - well, if not corrects, tests do catch this one.